apps/aecore/src/aec_parent_connector.erl

```diff
@@ -415,30 +420,30 @@ get_pinning_data() ->
     end.
 
 get_generations(FromHeight, ToHeight) ->
-    ReversedBlocks =
-        lists:foldl(
-            fun(Height, Accum) ->
-                case aec_chain:get_generation_by_height(Height, backward) of
-                    {ok, #{micro_blocks := MBs}} ->
-                        ReversedGeneration = lists:reverse(MBs),
-                        ReversedGeneration ++ Accum;
-                    error -> error({failed_to_fetch_generation, Height})
-                end
-            end,
-            [],
-            lists:seq(FromHeight, ToHeight)),
-    lists:reverse(ReversedBlocks).
+    lists:flatten(
+        lists:map(
+        fun(Height) ->
+            case aec_chain:get_generation_by_height(Height, backward) of
+                {ok, #{micro_blocks := MBs}} -> MBs;
+                error -> error({failed_to_fetch_generation, Height})
+            end
+        end,
+        lists:seq(FromHeight, ToHeight)
+    )).
 
 
 find_spends_to(Account) ->
     %% TODO Not happy with this. are we in the correct epoch when this is supposed to be 
     %%      called, and what if you want to go back through the chain and validate backwards?
     {ok, #{last := Last, first := First}} = aec_chain_hc:epoch_info(), 
     Blocks = get_generations(First, Last-1),
-    lists:flatten([ pick_pin_spends_to(Account, Txs) || {mic_block, _, Txs, _} <- Blocks ]).
+    lists:flatten([ pick_pin_spends_to(Account, aec_blocks:txs(B)) || B <- Blocks ]).
+
 
 pick_pin_spends_to(Account, Txs) ->
-    [ T || {signed_tx,{aetx,spend_tx,aec_spend_tx,_,{spend_tx,_,{id,account,Account2},_,_,_,_,T}},_} <- Txs, Account =:= Account2, is_pin(T)].
+    BareTxs = [ aetx_sign:tx(T) || T <- Txs],
+    InnerTxs = [ aetx:specialize_type(Tx) || Tx <- BareTxs,  spend_tx == aetx:tx_type(Tx)],
+    [ aec_spend_tx:payload(T) || {_, T} <- InnerTxs, aeser_id:specialize(aec_spend_tx:recipient_id(T)) == {account,Account}, is_pin(aec_spend_tx:payload(T))].
```

Comment:
Looks much better! 👍  
Alternatively the first to could perhaps be combined into:
```
SpendTxs = [ Tx || STx <- Txs, {spend_tx, Tx} <- [aetx:specialize_type(STx)] ],
```

Reply:
Yea, but double generators make my eyes hurt.